DOCSP-30783: certain interfaces are sealed in v4.10 (#411)
mongodb/docs-java: source/whats-new.txt

  aggregation framework and removed in a future release.
- Beginning with v5.0, the {+driver-short+} will require Java 11 or later.

.. _version-4.10:

What's New in 4.10
------------------

> - Interfaces in the ``com.mongodb.client.model.search`` package have the
>   ``@Sealed`` marking instead of ``@Evolving``. Sealed interfaces

Reviewer: Suggestion for brevity. If you want to stick with the current form, I'd call this an "annotation", not a "marking", to be consistent with the driver.

```suggestion
- Interfaces in the ``com.mongodb.client.model.search`` package are now
  ``@Sealed`` instead of ``@Evolving``. Sealed interfaces
```

>   ``@Sealed`` marking instead of ``@Evolving``. Sealed interfaces
>   restrict users from extending or implementing them.

Reviewer: Using language from the driver:

```suggestion
  ``@Sealed`` marking instead of ``@Evolving``. Sealed interfaces
  must not be extended or implemented by consumers of the library.
```
